lib/ansible/modules/cloud/amazon/aws_s3_bucket_facts.py

```diff
@@ -0,0 +1,147 @@
+short_description: Lists S3 buckets in AWS
+requirements:
+  - boto3 >= 1.4.4
+  - python >= 2.6
+description:
+    - Lists S3 buckets in AWS
+version_added: "2.4"
+author: "Gerben Geijteman (@hyperized)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Note: Only AWS S3 is currently supported
+
+# Lists all s3 buckets
+- aws_s3_bucket_facts:
+'''
+
+RETURN = '''
+buckets:
+  description: "List of buckets"
+  returned: always
+  sample:
+    - creation_date: 2017-07-06 15:05:12 +00:00
+      name: my_bucket
+  type: list
+owner:
+  description: "Details of bucket owner"
+  returned: always
+  sample:
+    - display_name: my_user
+      id: d5a0152fc8eb469fcc7eed35b2da939fd9e1a9c53924b681450f5353199900f8
+  type: dict
+response_metadata:
+  description: "Details of API call"
+  returned: always
+  sample:
+    - host_id: PoAM5+3hzaYKyNV/27cZcXkTW0eeZV8WyyJILUbzf/hEqtg7BcZW+sP9+7C8c6CFep8rbvFKPu4=
+      http_headers:
+        content-type: application/xml
+        date: "Mon, 31 Jul 2017 06:42:29 GMT"
+        server: AmazonS3
+        transfer-encoding: chunked
+        x-amz-id-2: PoAM5+3hzaYKyNV/27cZcXkTW0eeZV8WyyJILUbzf/hEqtg7BcZW+sP9+7C8c6CFep8rbvFKPu4=
+        x-amz-request-id: FB225485548AB650
+      http_status_code: 200
+      request_id: FB225485548AB650
+      retry_attempts: 0
+  type: dict
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, HAS_BOTO3, camel_dict_to_snake_dict, \
+    get_aws_connection_info
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+class S3ServiceManager(object):
+    """
+    Service manager for S3
+    """
+
+    def __init__(self, module):
+        """
+        Bootstrap module and connection
+        :param module:
+        """
+        self.module = module
+
+        # Set up connection
+        region, ec2_url, aws_connect_params = get_aws_connection_info(self.module, boto3=HAS_BOTO3)
+
+        if region:
+            try:
+                self.connection = boto3_conn(self.module, conn_type='client', resource='s3', region=region,
+                                             endpoint=ec2_url, **aws_connect_params)
+            except (botocore.exceptions.NoCredentialsError, botocore.exceptions.ProfileNotFound) as e:
+                self.module.fail_json(msg=e.message, exception=traceback.format_exc(),
+                                      **camel_dict_to_snake_dict(e.response))
+        else:
+            self.module.fail_json(msg="AWS region must be specified (like: us-east-1)")
+
+    def get_bucket_list(self):
+        """
+        Return result of list_buckets json encoded
+        :return:
+        """
+        try:
+            buckets = camel_dict_to_snake_dict(self.connection.list_buckets())
```

Comment:
We don't really want the response metadata in the returned facts, so this should probably have `['Buckets']` at the end.